From 35b0b99306423f7d01d2da9328d10dda2028fd9b Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Thu, 28 Apr 2022 07:31:34 +0300
Subject: [PATCH] [SW-79482] gaudi2: disable img validation for fake unsecurity

To support RMA debug in production, firmware needs support of
disabling security and image validation across the flow. In
case of ppboot validating preboot, it uses arcboot rom exported
functions to verify the image. For fake unsecurity, ppboot
should skip image validation completely.

Change-Id: I9cc1c7b28b54021e4e455880316fa4dedba85656
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 7b96330..6d7bd40 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -42,6 +42,7 @@
 #include <arcboot_export.h>
 #include <pci.h>
 #include <hl.h>
+#include <security.h>
 #include <address_space.h>
 #include <gaudi2/gaudi2_security.h>
 #include <tpm2.h>
@@ -374,10 +375,12 @@
 		image_size -= (CRC_SIZE * (num_cycles-1));
 
 	image_size -= SECUREBOOT_IMG_HEAD_FOOTER;
-	if (secureboot_verify_image(image_size,
-			(SDRAM_BASE + IMG_LEN_SIZE))
-			== SECUREBOOT_VERIFY_FAILED)
-		return ARCBOOT_BRINGUP_ERROR;
+	if (!preboot_security_fuse_override()) {
+		if (secureboot_verify_image(image_size,
+					    (SDRAM_BASE + IMG_LEN_SIZE))
+		    == SECUREBOOT_VERIFY_FAILED)
+			return ARCBOOT_BRINGUP_ERROR;
+	}
 
 	/* Hash is encrypted, and only first 48 bytes of it are used
 	 * Hash is read and not calculated because of ppboot image size
